shell/platform/linux/fl_key_event_plugin.cc

```diff
@@ -161,25 +161,31 @@ static uint64_t get_event_id(GdkEventKey* event) {
 
 // Finds an event in the event queue that was sent to the framework by its ID.
 static GdkEventKey* find_pending_event(FlKeyEventPlugin* self, uint64_t id) {
-  if (self->pending_events->len == 0 ||
-      FL_KEY_EVENT_PAIR(g_ptr_array_index(self->pending_events, 0))->id != id) {
-    return nullptr;
+  for (guint i = 0; i <= self->pending_events->len; ++i) {
+    if (FL_KEY_EVENT_PAIR(g_ptr_array_index(self->pending_events, i))->id ==
+        id) {
+      return FL_KEY_EVENT_PAIR(g_ptr_array_index(self->pending_events, 0))
```

Comment:
You're testing against index `i`, but then still returning index 0.

Reply:
Thanks for finding that.  I'll submit a fix today.